File: Wishlis.Domain/Entities/WishlistItem.cs
namespace Wishlis.Domain.Entities;

public class WishlistItem
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; }
    public double Price { get; set; }
    public string CurrencyCode { get; set; }
    public string Url { get; set; }
    public bool IsGroupGift { get; set; }
}

File: wishlis-webapp/src/wishlist/wishlistItem.ts

export interface WishlistItem {
  id: string;
  userId: string;
  name: string;
  price: number;
  currencyCode: string;
  url: string;
  isGroupGift: boolean;
}

export const createDefaultWishlistItem = (userId: string, currencyCode: string | undefined): WishlistItem => {
  return {
    id: crypto.randomUUID(),
    userId,
    name: "New Item",
    url: "",
    price: 0,
    currencyCode: currencyCode ?? "USD",
    isGroupGift: false
  }
}
